### DISCUSSION TITLE
Release notes for 1.56.0 - Templates now support Templater syntax

### DISCUSSION THREAD

**Discussion by danielo515** [danielo515/obsidian-modal-form#362] [Announcements]
## Templates now support Templater syntax

Modal Form 1.56.0 enhances its templates feature with Templater support!
I'm particularly excited about this integration because, with a little addition we are greatly improving not only the usability of this plugin, but also Templater.
Now binding a particular form to a complex template is easier than ever, knowing the fields available in the template also becomes a lot easier and I think the overall experience is greatly improved.

<img width="689" alt="image" src="https://github.com/user-attachments/assets/04ab85b3-924a-43fd-8050-e77f79be74b7" />

### What's New

The Templater integration adds the following capabilities to form templates:

- Use Templater syntax (`<% %>`) inside form templates
- Process dates, files, and system information through Templater
- Automatic processing of Templater syntax after form variables are replaced

Here's a little example combining both syntaxes:

```
---
created: <% tp.date.now() %>
---

Dear {{name}},

Meeting scheduled for <% tp.date.now("MMMM Do, YYYY") %>
Location: {{location}}

Best regards,
{{signature}}
```

### How it Works

The template processing now happens in two stages:

1. First, Modal Form replaces all form variables (the `{{variable}}` syntax)
2. Then, if Templater is available, it processes any Templater syntax in the resulting text

### Retry Functionality

Getting templater templates right at the first try can be hard, that's why we added a retry option when templater processing part fails.
If something goes wrong during templater processing , Modal Form now provides a retry modal where you can try to fix the template and try again.:

<img width="591" alt="image" src="https://github.com/user-attachments/assets/dd82dcea-2523-4a8c-a554-e3d240b05485" />

This makes it easier to fix any issues without losing your form data and gives you an insight on how te template is processed.

Check out the [templates documentation](/docs/templates) for more details and examples of how to use this new feature.

**Comment by danielo515** [comment]
@motion2082 yo may find this one interesting

**Reply by motion2082** [reply]
Nice work Danielo515, this feature will definitely come in handy for Modal Form users
